zk-sdk/src/encryption/elgamal.rs

```diff
@@ -145,17 +153,33 @@ pub struct ElGamalKeypair {
     secret: ElGamalSecretKey,
 }
 
+#[wasm_bindgen]
+impl ElGamalKeypair {
+    /// Generates the public and secret keys for ElGamal encryption.
+    ///
+    /// This function is randomized. It internally samples a scalar element using `OsRng`.
+    pub fn new_rand() -> Self {
+        ElGamal::keygen()
+    }
+
+    pub fn pubkey_owned(&self) -> ElGamalPubkey {
+        self.public
+    }
+}
+
 impl ElGamalKeypair {
```

Comment:
nit: Would it make more sense to have this whole impl block under `#[cfg(not(target_arch = "wasm32"))]`?

Reply:
Yep good idea. Done!